staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go

```diff
@@ -18,13 +18,19 @@ package etcd3
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 
 	"github.com/coreos/etcd/clientv3"
+	"google.golang.org/grpc/grpclog"
 	"k8s.io/klog"
 )
 
 func init() {
 	clientv3.SetLogger(klogWrapper{})
+
+	// overwrite gRPC logger, to discard all gRPC info-level logging
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, os.Stderr, os.Stderr))
```

Comment:
I didn't expect to send things directly to os.Stderr... that doesn't get seen by things that expected to direct error logs to a file

Reply:
Is log writer configurable in kube-apiserver? If we expose log writer from `klog`, so we emit logs to the same writer for warning and error messages from gRPC. Or pase `--logtostderr` flag?

Reply:
As Gyuho pointed offline, it seems like this is similar to what we're doing in etcd server as well - https://github.com/etcd-io/etcd/blob/bf3d1fb48429a0050f82b5b3233d86903009b327/embed/config_logging.go#L104

Reply:
> Is log writer configurable in kube-apiserver?

Yes, via klog flags

Do we have other grpc clients that would be affected by this? cc @immutableT for storage encryption client

Reply:
Seems like some of the fields need to be exported if we want to keep the same writer

```go

// loggingT collects all the global state of the logging setup.
type loggingT struct {
	// Boolean flags. Not handled atomically because the flag.Value interface
	// does not let us avoid the =true, and that shorthand is necessary for
	// compatibility. TODO: does this matter enough to fix? Seems unlikely.
	toStderr     bool // The -logtostderr flag.
	alsoToStderr bool // The -alsologtostderr flag.
        ...
	// file holds writer for each of the log types.
	file [numSeverity]flushSyncWriter
```

https://github.com/kubernetes/klog/blob/d256e8871dc002fa06b16e040f24c3278dd638d3/klog.go#L443-L465

Reply:
Instead of exporting these fields, you could add an exported method that could be used to retrieve this information for callers that need to determine log configuration.

Alternately, you could also add this override directly into the klog package itself, possible with some flag to control the override? What do you think @dims?

Reply:
[Loggerv2](https://pkg.go.dev/google.golang.org/grpc/grpclog#LoggerV2) is an interface, why can we not also use a klog wrapper here?